src/planwise/client/scenarios/views.cljs

```diff
@@ -22,17 +22,30 @@
     (str "<b>" (utils/escape-html name) "</b><br> Capacity: " capacity)
     (str "<b> New provider " (:index provider) "</b><br> Click on panel for editing... ")))
 
+(defn- show-source
+  [{{:keys [name quantity quantity-current]} :elem :as source}]
+  (str "<b>" (utils/escape-html name) "</b>"
+       "<br> Original quantity: " quantity
+       "<br> Current quantity: " quantity-current))
+
+(defn- to-indexed-map
+  [coll]
+  (map-indexed (fn [idx elem] {:elem elem :index idx}) coll))
+
 (defn simple-map
   [{:keys [bbox]} scenario]
   (let [state    (subscribe [:scenarios/view-state])
         index    (subscribe [:scenarios/changeset-index])
         position (r/atom mapping/map-preview-position)
         zoom     (r/atom 3)
         add-point (fn [lat lon] (dispatch [:scenarios/create-provider {:lat lat
-                                                                       :lon lon}]))]
+                                                                       :lon lon}]))
+        use-providers-clustering false
+        providers-type-layer     (if use-providers-clustering :cluster-layer :point-layer)]
     (fn [{:keys [bbox]} {:keys [changeset providers raster] :as scenario}]
-      (let [providers           (into providers changeset)
-            indexed-providers   (map-indexed (fn [idx elem] {:elem elem :index idx}) providers)
+      (let [providers             (into providers changeset)
+            indexed-providers     (to-indexed-map providers)
+            indexed-sources       (to-indexed-map (:sources scenario))
```

Comment:
You don't need to index sources. In fact, indexing providers for identification later when editing is wrong as well, although pre-existing.